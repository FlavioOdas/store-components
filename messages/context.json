{
  "store/availability-subscriber.added-message": "E-mail registered successfully",
  "store/availability-subscriber.email-placeholder": "Email",
  "store/availability-subscriber.error-message": "Failed, please try again",
  "store/availability-subscriber.invalid-email": "Invalid email",
  "store/availability-subscriber.name-placeholder": "Name",
  "store/availability-subscriber.send-label": "Send",
  "store/availability-subscriber.subscribe-label": "I want to know when this product becomes available",
  "store/availability-subscriber.title": "This product isn't available right now",
  "store/buy-button.add-to-cart": "Add to cart",
  "store/buyButton-label-unavailable": "Unavailable",
  "store/buybutton.add-failure": "Something went wrong and the item wasn't added to your cart!",
  "store/buybutton.buy-offline-success": "Item added to your offline cart",
  "store/buybutton.buy-success": "Item added to your cart",
  "store/buybutton.buy-success-duplicate": "Item already in your cart",
  "store/buybutton.see-cart": "View cart",
  "store/buybutton.select-sku-variations": "Please select an option of each variation",
  "store/greeting": "Hello",
  "store/loading": "Loadingâ€¦",
  "store/newsletter.confirmationText": "Your email was successfully registered.",
  "store/newsletter.confirmationTitle": "Thank you!",
  "store/newsletter.error": "Something went wrong, please try again later",
  "store/newsletter.invalidEmail": "Please, enter a valid email.",
  "store/newsletter.label": "Subscribe to our newsletter",
  "store/newsletter.placeholder": "Enter your email address",
  "store/newsletter.submit": "Sign up",
  "store/pricing.from": "From",
  "store/pricing.installment-display": "Or {installments} {times} of {installmentPrice}",
  "store/pricing.interest-free": "interest-free",
  "store/pricing.savings": "Save {savings}",
  "store/pricing.to": "To",
  "store/product-description.collapse.showLess": "Show less",
  "store/product-description.collapse.showMore": "Show more",
  "store/product-description.property": "Property",
  "store/product-description.specification": "Specification",
  "store/product-description.title": "Product Description",
  "store/search.placeholder": "Search",
  "store/search.search-term-too-short": "Search term is too short",
  "store/search.searchFor": "Search for {term}",
  "store/searchBar.button.cancel.label": "Cancel",
  "store/share.title": "{product} {sku} at {store}: ",
  "admin/editor.info-card.title": "Title of InfoCard component",
  "admin/editor.info-card.description": "Description of InfoCard component",
  "admin/editor.info-card.isFullModeStyle.title": "Title of isFullModeStyle property",
  "admin/editor.info-card.isFullModeStyle.description": "Description of isFullModeStyle property",
  "admin/editor.info-card.textPosition.title": "Title of textPosition property",
  "admin/editor.info-card.textPosition.description": "Description of textPosition property",
  "admin/editor.info-card.headline.title": "Title of headline property",
  "admin/editor.info-card.headline.description": "Description of headline property",
  "admin/editor.info-card.subhead.title": "Title of subhead property",
  "admin/editor.info-card.subhead.description": "Description of subhead property",
  "admin/editor.info-card.callToActionMode.title": "Title of callToActionMode property",
  "admin/editor.info-card.callToActionMode.description": "Description of callToActionMode property",
  "admin/editor.info-card.callToActionText.title": "Title of callToActionText property",
  "admin/editor.info-card.callToActionText.description": "Description of callToActionText property",
  "admin/editor.info-card.callToActionUrl.title": "Title of callToActionUrl property",
  "admin/editor.info-card.callToActionUrl.description": "Description of callToActionUrl property",
  "admin/editor.info-card.imageUrl.title": "Title of imageUrl property",
  "admin/editor.info-card.imageUrl.description": "Description of imageUrl property",
  "admin/editor.info-card.mobileImageUrl.title": "Title of mobileImageUrl property",
  "admin/editor.info-card.mobileImageUrl.description": "Description of mobileImageUrl property",
  "admin/editor.info-card.textAlignment.title": "Title of textAlignment property",
  "admin/editor.info-card.textAlignment.description": "Description of textAlignment property",
  "admin/editor.info-card.textPosition.left": "Name of enum value to make text at left position",
  "admin/editor.info-card.textPosition.center": "Name of enum value to make text at center position",
  "admin/editor.info-card.textPosition.right": "Name of enum value to make text at right position",
  "admin/editor.info-card.textAlignment.left": "Name of enum value to align text to the left",
  "admin/editor.info-card.textAlignment.center": "Name of enum value to align text to the center",
  "admin/editor.info-card.textAlignment.right": "Name of enum value to align text to the right",
  "admin/editor.info-card.callAction.none": "Name of enum value to make CTA disappear",
  "admin/editor.info-card.callAction.button": "Name of enum value to make CTA a button",
  "admin/editor.info-card.callAction.link": "Name of enum value to make CTA a link",
  "admin/editor.info-card.imageActionUrl.title": "Name of imageActionUrl property",
  "admin/editor.info-card.imageActionUrl.description": "Name of imageActionUrl description",
  "store/user-address.pickup": "Pickup at <address>",
  "store/user-address.order": "Order to <address>",
  "store/user-address.change": "Change address button label",
  "admin/editor.notification-bar.title": "Notification bar",
  "admin/editor.notification-bar.description": "A component that show any text in a bar style",
  "admin/editor.notification-bar.content.title": "Content",
  "admin/editor.notification-bar.content.description": "Text of the bar",
  "admin/editor.blockClass.title": "CSS Block Class",
  "admin/editor.blockClass.description": "Adds an extra class name to ease styling",
  "admin/editor.image.title": "Image",
  "admin/editor.image.description": "Show any image",
  "admin/editor.image.src.title": "Image",
  "admin/editor.image.title.title": "Image title",
  "admin/editor.image.title.description": "Title to be shown on hover",
  "admin/editor.image.alt.title": "Alternative text",
  "admin/editor.product-images.title": "ProductImages component title on storefront",
  "admin/editor.product-images.description": "Description of ProductImages component",
  "admin/editor.product-images.zoomOptions.title": "Title of the Zoom options provided by the product images app",
  "admin/editor.product-images.zoomType.title": "Title of the zoomType option",
  "admin/editor.product-images.bgopacity.title": "Title of the background opacity of the gallery when zoom is set to in gallery mode",
  "admin/editor.product-images.thumbnailsOrientation.title": "Title of the thumbnailsOrientation option",
  "admin/editor.product-images.gallery": "Used in 'gallery' zoom option",
  "admin/editor.product-images.in-page": "Used in 'gallery' zoom option",
  "admin/editor.product-images.no-zoom": "Used in 'gallery' zoom option",
  "admin/editor.product-images.vertical": "Used in 'direction' option, as vertical orientation",
  "admin/editor.product-images.horizontal": "Used in 'direction' option, as horizontal orientation",
  "admin/editor.product-images.position.title": "Title of the position option",
  "admin/editor.product-images.position.description": "Description of the position option",
  "admin/editor.product-images.left": "Used in 'position' option",
  "admin/editor.product-images.right": "Used in 'position' option",
  "admin/editor.product-images.displayThumbnailsArrows.title": "Title of the displayThumbnailsArrows option",
  "admin/editor.product-specifications.title": "Product Specifications",
  "admin/editor.product-specifications.hidden-specifications.title": "Hidden Specficiations",
  "admin/editor.product-specifications.hidden-specifications.description": "Type names of specifications you want to hide",
  "admin/editor.product-specifications.visible-specifications.title": "Visible Specifications",
  "admin/editor.product-specifications.visible-specifications.description": "Type names of specifications you want to appear. Only provide one of the hidden specifications or visible specifications",
  "admin/editor.product-specifications.items.title": "Specification",
  "admin/editor.logo.title": "Name of Logo Component",
  "admin/editor.logo.description": "Description of Logo Component",
  "admin/editor.logo.href.title": "Title of href property of Logo",
  "admin/editor.logo.href.description": "Description of href property of Logo",
  "admin/editor.logo.url.title": "Title of url property of Logo",
  "admin/editor.logo.url.description": "Description of url property of Logo",
  "admin/editor.logo.width.title": "Title of width property of Logo",
  "admin/editor.logo.width.description": "Description of width property of Logo",
  "admin/editor.logo.height.title": "Title of height property of Logo",
  "admin/editor.logo.height.description": "Description of height property of Logo",
  "admin/editor.logo.title.title": "Title of title property of Logo",
  "admin/editor.logo.title.description": "Description of title property of Logo",
  "admin/editor.logo.mobileWidth.title": "Title of mobileWidth property of Logo",
  "admin/editor.logo.mobileWidth.description": "Description of mobileWidth property of Logo",
  "admin/editor.logo.mobileHeight.title": "Title of mobileHeight property of Logo",
  "admin/editor.logo.mobileHeight.description": "Description of mobileHeight property of Logo",
  "store/skuSelector.seeMore": "Press button to see remaining items",
  "store/sku-selector.select.placeholder": "Choose an option",
  "store/sku-selector.variation.select-an-option": "You have to select an option",
  "admin/editor.skuSelector.title": "Title of SKU Selector component",
  "admin/editor.skuSelector.description": "Description of SKU Selector component",
  "admin/editor.skuSelector.seeMoreLabel.title": "Title of seeMoreLabel property",
  "admin/editor.skuSelector.seeMoreLabel.description": "Description of seeMoreLabel property",
  "admin/editor.product-description.title": "Product description",
  "admin/editor.product-description.title-prop.title": "Title",
  "admin/editor.search-bar.title": "Search bar",
  "admin/editor.search-bar.placeholder.title": "Placeholder",
  "store/shipping.empty-sla": "There's no shipping information available for the informed zip code",
  "store/shipping.free": "Free",
  "store/shipping.label": "Calculate shipping",
  "store/store-components.share.label": "Share",
  "store/technicalspecifications.title": "Technical Specifications"
}