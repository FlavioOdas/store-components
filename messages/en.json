{
  "store/availability-subscriber.added-message": "E-mail registered successfully",
  "store/availability-subscriber.email-placeholder": "Email",
  "store/availability-subscriber.error-message": "Failed, please try again",
  "store/availability-subscriber.invalid-email": "Invalid email",
  "store/availability-subscriber.name-placeholder": "Name",
  "store/availability-subscriber.send-label": "Send",
  "store/availability-subscriber.subscribe-label": "I want to know when this product becomes available",
  "store/availability-subscriber.title": "This product isn't available right now",
  "store/buy-button.add-to-cart": "Add to cart",
  "store/buyButton-label-unavailable": "Unavailable",
  "store/buybutton.add-failure": "Something went wrong and the item wasn't added to your cart!",
  "store/buybutton.buy-offline-success": "Item added to your offline cart",
  "store/buybutton.buy-success": "Item added to your cart",
  "store/buybutton.buy-success-duplicate": "Item already in your cart",
  "store/buybutton.see-cart": "View cart",
  "store/buybutton.select-sku-variations": "Please select an option of each variation",
  "store/greeting": "Hello",
  "store/loading": "Loadingâ€¦",
  "store/newsletter.confirmationText": "Your email was successfully registered.",
  "store/newsletter.confirmationTitle": "Thank you!",
  "store/newsletter.error": "Something went wrong, please try again later",
  "store/newsletter.invalidEmail": "Please, enter a valid email.",
  "store/newsletter.label": "Subscribe to our newsletter",
  "store/newsletter.placeholder": "Enter your email address",
  "store/newsletter.submit": "Sign up",
  "store/pricing.from": "From",
  "store/pricing.installment-display": "Or {installments} {times} of {installmentPrice}",
  "store/pricing.interest-free": "interest-free",
  "store/pricing.savings": "Save {savings}",
  "store/pricing.to": "To",
  "store/product-description.collapse.showLess": "Show less",
  "store/product-description.collapse.showMore": "Show more",
  "store/product-description.property": "Property",
  "store/product-description.specification": "Specification",
  "store/product-description.title": "Product Description",
  "store/search.placeholder": "Search",
  "store/search.search-term-too-short": "Search term is too short",
  "store/search.searchFor": "Search for {term}",
  "store/searchBar.button.cancel.label": "Cancel",
  "store/share.title": "{product} {sku} at {store}: ",
  "admin/editor.info-card.title": "InfoCard",
  "admin/editor.info-card.description": "A component with image and text that you can choose how to display them",
  "admin/editor.info-card.isFullModeStyle.title": "Image as background",
  "admin/editor.info-card.isFullModeStyle.description": "If true, image provided will be used as a background image and text will be displayed over it",
  "admin/editor.info-card.textPosition.title": "Text Position",
  "admin/editor.info-card.textPosition.description": "Choose in which position of the component text will be displayed, left, center or right",
  "admin/editor.info-card.headline.title": "Headline",
  "admin/editor.info-card.headline.description": "Text to be displayed as a headline inside component",
  "admin/editor.info-card.subhead.title": "Subhead",
  "admin/editor.info-card.subhead.description": "Text to be displayed underneath headline inside component",
  "admin/editor.info-card.callToActionMode.title": "Call to Action Mode",
  "admin/editor.info-card.callToActionMode.description": "Controls how your call to action is displayed",
  "admin/editor.info-card.callToActionText.title": "Call to Action text",
  "admin/editor.info-card.callToActionText.description": "Text to show inside call to action component",
  "admin/editor.info-card.callToActionUrl.title": "Call to Action URL",
  "admin/editor.info-card.callToActionUrl.description": "Url to redirect to when CTA is clicked",
  "admin/editor.info-card.imageUrl.title": "Image",
  "admin/editor.info-card.imageUrl.description": "Image to be used on desktop",
  "admin/editor.info-card.mobileImageUrl.title": "Mobile Image",
  "admin/editor.info-card.mobileImageUrl.description": "Image to be used on mobile",
  "admin/editor.info-card.textAlignment.title": "Text Alignment",
  "admin/editor.info-card.textAlignment.description": "Control the text alignment inside component",
  "admin/editor.info-card.textPosition.left": "Left",
  "admin/editor.info-card.textPosition.center": "Center",
  "admin/editor.info-card.textPosition.right": "Right",
  "admin/editor.info-card.textAlignment.left": "Left",
  "admin/editor.info-card.textAlignment.center": "Center",
  "admin/editor.info-card.textAlignment.right": "Right",
  "admin/editor.info-card.callAction.none": "None",
  "admin/editor.info-card.callAction.button": "Button",
  "admin/editor.info-card.callAction.link": "Link",
  "admin/editor.info-card.imageActionUrl.title": "Image Action URL",
  "admin/editor.info-card.imageActionUrl.description": "Pass URL to make image a link and able to redirect user",
  "store/user-address.pickup": "Pick up at {name}",
  "store/user-address.order": "Order to",
  "store/user-address.change": "Change",
  "admin/editor.notification-bar.title": "Notification bar",
  "admin/editor.notification-bar.description": "A component that show any text in a bar style",
  "admin/editor.notification-bar.content.title": "Content",
  "admin/editor.notification-bar.content.description": "Text of the bar",
  "admin/editor.blockClass.title": "CSS Block Class",
  "admin/editor.blockClass.description": "Adds an extra class name to ease styling",
  "admin/editor.image.title": "Image",
  "admin/editor.image.description": "Show any image",
  "admin/editor.image.src.title": "Image",
  "admin/editor.image.title.title": "Image title",
  "admin/editor.image.title.description": "Title to be shown on hover",
  "admin/editor.image.alt.title": "Alternative text",
  "admin/editor.product-images.title": "Product Images",
  "admin/editor.product-images.description": "Product images component allow you to customize the product images",
  "admin/editor.product-images.zoomOptions.title": "Zoom Options",
  "admin/editor.product-images.zoomType.title": "Zoom Type",
  "admin/editor.product-images.bgopacity.title": "Background opacity",
  "admin/editor.product-images.thumbnailsOrientation.title": "Thumbnails Orientation",
  "admin/editor.product-images.gallery": "In Gallery",
  "admin/editor.product-images.in-page": "In page",
  "admin/editor.product-images.no-zoom": "No zoom",
  "admin/editor.product-images.vertical": "Vertical",
  "admin/editor.product-images.horizontal": "Horizontal",
  "admin/editor.product-images.position.title": "Thumbnail position",
  "admin/editor.product-images.position.description": "Used only when orientation is vertical",
  "admin/editor.product-images.left": "Left",
  "admin/editor.product-images.right": "Right",
  "admin/editor.product-images.displayThumbnailsArrows.title": "Display Arrows in Thumbnails",
  "admin/editor.product-specifications.title": "Product Specifications",
  "admin/editor.product-specifications.hidden-specifications.title": "Hidden Specficiations",
  "admin/editor.product-specifications.hidden-specifications.description": "Type names of specifications you want to hide",
  "admin/editor.product-specifications.visible-specifications.title": "Visible Specifications",
  "admin/editor.product-specifications.visible-specifications.description": "Type names of specifications you want to appear. Only provide one of the hidden specifications or visible specifications",
  "admin/editor.product-specifications.items.title": "Specification",
  "admin/editor.logo.title": "Logo",
  "admin/editor.logo.description": "Component to show your store logo",
  "admin/editor.logo.href.title": "Link to redirect",
  "admin/editor.logo.href.description": "Link to be redirected when logo is clicked. Leave empty to make it not clickable.",
  "admin/editor.logo.url.title": "Image",
  "admin/editor.logo.url.description": "Image to be used",
  "admin/editor.logo.width.title": "Width",
  "admin/editor.logo.width.description": "Width to be used for logo.",
  "admin/editor.logo.height.title": "Height",
  "admin/editor.logo.height.description": "Height to be used for logo.",
  "admin/editor.logo.title.title": "Alternative text",
  "admin/editor.logo.title.description": "Alternative text to be provided to logo image",
  "admin/editor.logo.mobileWidth.title": "Mobile Width",
  "admin/editor.logo.mobileWidth.description": "Width to be used on mobile. If empty, will use the provided width value.",
  "admin/editor.logo.mobileHeight.title": "Mobile Height",
  "admin/editor.logo.mobileHeight.description": "Height to be used on mobile. If empty, will use the provided height value.",
  "store/shipping.empty-sla": "There's no shipping information available for the informed zip code",
  "store/shipping.free": "Free",
  "store/shipping.label": "Calculate shipping",
  "store/sku-selector.select.placeholder": "Choose an option",
  "store/sku-selector.variation.select-an-option": "Select an option",
  "store/skuSelector.seeMore": "See {quantity} more",
  "store/store-components.share.label": "Share",
  "store/technicalspecifications.title": "Technical Specifications"
}