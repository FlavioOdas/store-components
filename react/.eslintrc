{
  "rules": {
    "react/jsx-filename-extension": "off",
    "@typescript-eslint/ban-ts-comment": 0
  },
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "overrides": [
    {
      "files": ["*.js", "*.jsx"],
      "parser": "babel-eslint",
      "parserOptions": {
        "sourceType": "module"
      }
    }
  ]
}
