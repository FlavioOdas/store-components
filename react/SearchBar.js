import SearchBar from './components/SearchBar/index'
//teste

export default SearchBar
