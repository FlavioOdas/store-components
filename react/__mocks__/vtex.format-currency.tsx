import React, { Fragment } from 'react'
import { useRuntime } from 'vtex.render-runtime'
import { injectIntl } from 'react-intl'

export function formatCurrency({ intl, culture, value }: any) {
  return intl.formatNumber(value, {
    style: 'currency',
    currency: culture.currency,
    ...(culture.customCurrencyDecimalDigits != null
      ? { minimumFractionDigits: culture.customCurrencyDecimalDigits }
      : {}),
  })
}

function BaseFormattedCurrency({ value, intl }: any) {
  const { culture } = useRuntime()

  const number = intl.formatNumber(value, {
    style: 'currency',
    currency: culture.currency,
    ...(culture.customCurrencyDecimalDigits != null
      ? { minimumFractionDigits: culture.customCurrencyDecimalDigits }
      : {}),
  })

  return <Fragment>{number}</Fragment>
}

export const FormattedCurrency = injectIntl(BaseFormattedCurrency)
