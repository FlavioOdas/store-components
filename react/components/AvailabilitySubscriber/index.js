import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ApolloConsumer } from 'react-apollo'
import { defineMessages, injectIntl } from 'react-intl'
import { Button, Input } from 'vtex.styleguide'

import ADD_TO_AVAILABILITY_SUBSCRIBER_MUTATION from './mutations/addToAvailabilitySubscriberMutation.gql'
import styles from './styles.css'

const messages = defineMessages({
  availabilitySubscriberInvalidEmail: {
    id: 'store/availability-subscriber.invalid-email',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberAddedMessage: {
    id: 'store/availability-subscriber.added-message',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberTitle: {
    id: 'store/availability-subscriber.title',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberSubscribeLabel: {
    id: 'store/availability-subscriber.subscribe-label',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberNamePlaceholder: {
    id: 'store/availability-subscriber.name-placeholder',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberEmailPlaceholder: {
    id: 'store/availability-subscriber.email-placeholder',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberSendLabel: {
    id: 'store/availability-subscriber.send-label',
    from: 'vtex.store-messages',
  },
  availabilitySubscriberErrorMessage: {
    id: 'store/availability-subscriber.error-message',
    from: 'vtex.store-messages',
  },
})

/**
 * Represents the availability subscriber form, that's shown when
 * the product isn't available.
 */
class AvailabilitySubscriber extends Component {
  emailInput = React.createRef()
  nameInput = React.createRef()

  state = {
    name: '',
    email: '',
    emailError: '',
    hasBlurredEmail: false,
    isLoading: false,
    sendStatus: '',
  }

  static propTypes = {
    /** The id of the current product sku */
    skuId: PropTypes.string.isRequired,
  }

  validateEmail = email => {
    const { emailError } = this.state

    const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

    let error = ''

    if (!emailRegex.test(email.toLowerCase())) {
      error = messages.availabilitySubscriberInvalidEmail.id
    }

    if (error !== emailError) {
      this.setState({
        emailError: error,
      })
    }
  }

  handleEmailBlur = () => {
    if (this.state.hasBlurredEmail) {
      return
    }

    this.setState({
      hasBlurredEmail: true,
    })
  }

  handleInputChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    })

    if (e.target.name === 'email') {
      this.validateEmail(e.target.value)
    }
  }

  handleSubmit = (e, client) => {
    e.preventDefault()
    const variables = {
      acronym: 'AS',
      document: {
        fields: [
          {
            key: 'skuId',
            value: this.props.skuId,
          },
          {
            key: 'name',
            value: this.state.name,
          },
          {
            key: 'email',
            value: this.state.email,
          },
          {
            key: 'notificationSend',
            value: 'false',
          },
          {
            key: 'createdAt',
            value: new Date().toISOString(),
          },
          {
            key: 'sendAt',
            value: null,
          },
        ],
      },
    }
    this.setState({
      isLoading: true,
    })
    client
      .mutate({
        mutation: ADD_TO_AVAILABILITY_SUBSCRIBER_MUTATION,
        variables,
      })
      .then(
        mutationRes => {
          this.setState({
            name: '',
            email: '',
            isLoading: false,
            sendStatus: 'success',
          })
        },
        mutationErr => {
          console.log('ERROR: ', mutationErr)
          this.setState({
            isLoading: false,
            sendStatus: 'error',
          })
        }
      )
    const event = new Event('message:success')
    event.details = {
      success: true,
      message: this.translate(messages.availabilitySubscriberAddedMessage.id),
    }
    document.dispatchEvent(event)
  }

  translate = id => this.props.intl.formatMessage({ id })

  componentDidMount() {
    this.setState({
      email: this.emailInput.value || '',
      name: this.nameInput.value || '',
    })
  }

  render() {
    const {
      name,
      email,
      emailError,
      hasBlurredEmail,
      isLoading,
      sendStatus,
    } = this.state

    const isFormDisabled =
      name === '' || email === '' || emailError !== '' || isLoading

    let emailErrorMessage = ''

    if (hasBlurredEmail && emailError !== '') {
      emailErrorMessage = this.translate(emailError)
    }

    return (
      <ApolloConsumer>
        {client => (
          <div className={styles.subscriberContainer}>
            <div className={`${styles.title} t-body mb3`}>
              {this.translate(messages.availabilitySubscriberTitle.id)}
            </div>
            <div className={`${styles.subscribeLabel} t-small fw3`}>
              {this.translate(messages.availabilitySubscriberSubscribeLabel.id)}
            </div>
            <form
              className={`${styles.form} mb4`}
              onSubmit={e => this.handleSubmit(e, client)}
            >
              <div
                className={`${styles.content} flex-ns justify-between mt4 mw6`}
              >
                <div
                  className={`${styles.input} ${styles.inputName} w-100 mr5 mb4`}
                >
                  <Input
                    name="name"
                    type="text"
                    placeholder={this.translate(
                      messages.availabilitySubscriberNamePlaceholder.id
                    )}
                    value={name}
                    onChange={this.handleInputChange}
                    ref={e => {
                      this.nameInput = e
                    }}
                  />
                </div>
                <div
                  className={`${styles.input} ${styles.inputEmail} w-100 mr5 mb4`}
                >
                  <Input
                    name="email"
                    type="text"
                    placeholder={this.translate(
                      messages.availabilitySubscriberEmailPlaceholder.id
                    )}
                    value={email}
                    onChange={this.handleInputChange}
                    onBlur={this.handleEmailBlur}
                    error={hasBlurredEmail && !!emailError}
                    errorMessage={emailErrorMessage}
                    ref={e => {
                      this.emailInput = e
                    }}
                  />
                </div>
                <div className={`${styles.submit} flex items-center mb4`}>
                  <Button
                    type="submit"
                    variation="primary"
                    size="small"
                    disabled={isFormDisabled}
                    isLoading={isLoading}
                  >
                    {this.translate(
                      messages.availabilitySubscriberSendLabel.id
                    )}
                  </Button>
                </div>
              </div>
              {sendStatus === 'success' && (
                <div className={`${styles.success} t-body c-success`}>
                  {this.translate(
                    messages.availabilitySubscriberAddedMessage.id
                  )}
                </div>
              )}
              {sendStatus === 'error' && (
                <div className={`${styles.error} c-danger`}>
                  {this.translate(
                    messages.availabilitySubscriberErrorMessage.id
                  )}
                </div>
              )}
            </form>
          </div>
        )}
      </ApolloConsumer>
    )
  }
}

export default injectIntl(AvailabilitySubscriber)
