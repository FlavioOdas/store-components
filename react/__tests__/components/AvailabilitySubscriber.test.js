import React from 'react'
import { render } from '@vtex/test-tools/react'

import AvailabilitySubscriber from '../../AvailabilitySubscriber'
import { messages } from '../../__mocks__/messages'

describe('<AvailabilitySubscriber />', () => {
  function renderComponent(customProps = {}) {
    const props = {
      skuId: '111111',
      available: false,
      ...customProps,
    }

    return render(<AvailabilitySubscriber {...props} />, {
      graphql: { mocks: [] },
      messages,
    })
  }

  it('should be able to mount and not break', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
