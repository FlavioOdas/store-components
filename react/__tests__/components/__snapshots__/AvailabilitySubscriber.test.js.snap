// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabilitySubscriber /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="subscriberContainer"
  >
    <div
      class="title t-body mb3"
    >
      store/availability-subscriber.title
    </div>
    <div
      class="subscribeLabel t-small fw3"
    >
      store/availability-subscriber.subscribe-label
    </div>
    <form
      class="form mb4"
    >
      <div
        class="content flex-ns justify-between mt4 mw6"
      >
        <div
          class="input inputName w-100 mr5 mb4"
        >
          <label>
            <input
              name="name"
              placeholder="store/availability-subscriber.name-placeholder"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="input inputEmail w-100 mr5 mb4"
        >
          <label>
            <input
              data-error="false"
              data-errormessage=""
              name="email"
              placeholder="store/availability-subscriber.email-placeholder"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="submit flex items-center mb4"
        >
          <button
            data-isloading="false"
            data-variation="primary"
            disabled=""
            type="submit"
          >
            store/availability-subscriber.send-label
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
